### ISSUE TITLE
Add in edge layer to adapt palyload to proper contract

### ISSUE THREAD

**Issue by ralphg6** [ralphg6/david#19]
The main idea is to support API payloads of any platforms. By example:

For `input` in Watson's API payload format:
```
POST http://localhost:5005/dialog?_if=watson
```
If the input format isn't explicitly informed, maybe we can build a component that tries to resolve the format. 

Also maybe we can implement support to `output format`, but only if will be needed.

**Comment by ralphg6** [comment]
I'm working in this functionality in branch: [issue/19-edge-adapter](https://github.com/ralphg6/david/tree/issue/19-edge-adapter)

**Comment by ralphg6** [comment]
During the development, I realized that not have a sense parametrize a different format in input and output.
So I implemented the support for 'adapter' query param that defines the format of input payload and output payload.

The usage is:
```
http://localhost:5000/dialog?adapter=watson
```
Where "watson" is the name of the adapter.

Until the moment, only "message" adapter is working. And he is the default adapter.

I will close this issue and create two new to develop the Watson adapter and to create a config for change default adapter of running instance.
